pkg/app/middlewares/instance_middleware.go

```diff
@@ -0,0 +1,60 @@
+package middlewares
+
+import (
+	"context"
+	"strings"
+
+	"github.com/aserto-dev/aserto-grpc/grpcutil"
+	"github.com/aserto-dev/go-authorizer/aserto/authorizer/v2"
+	"github.com/aserto-dev/go-authorizer/aserto/authorizer/v2/api"
+	"github.com/aserto-dev/topaz/pkg/cc/config"
+	grpcmiddleware "github.com/grpc-ecosystem/go-grpc-middleware"
+	"github.com/rs/zerolog"
+	"google.golang.org/grpc"
+)
+
+type InstanceMiddleware struct {
+	policyName    string
+	instanceLabel string
+	logger        *zerolog.Logger
+}
+
+func NewInstanceMiddleware(cfg *config.Config, logger *zerolog.Logger) *InstanceMiddleware {
+	details := strings.Split(*cfg.OPA.Config.Discovery.Resource, "/")
+
+	return &InstanceMiddleware{
+		policyName:    details[0],
+		instanceLabel: details[1],
+		logger:        logger,
+	}
+}
+
+var _ grpcutil.Middleware = &InstanceMiddleware{}
+
+// If the unary operation is an Is request attach configured instance information to request.
+func (m *InstanceMiddleware) Unary() grpc.UnaryServerInterceptor {
+	return func(ctx context.Context, req interface{}, info *grpc.UnaryServerInfo, handler grpc.UnaryHandler) (interface{}, error) {
+		request, ok := req.(*authorizer.IsRequest)
+		if ok {
+			request.PolicyInstance = &api.PolicyInstance{
+				InstanceLabel: m.instanceLabel,
+				Name:          m.policyName,
+			}
+			result, err := handler(ctx, req)
```

Comment:
You don't need to call the handler and return in this block. Just overwrite `request.PolicyInstance` and exit the `if` block.